Lib/test/test_subprocess.py

```diff
@@ -1360,6 +1361,26 @@ def test_communicate_epipe(self):
         self.addCleanup(p.stdin.close)
         p.communicate(b"x" * 2**20)
 
+    def test_string_representation(self):
+        """Test string representation Popen object."""
+        code = 'import sys; input(); sys.exit(57)'
+        cmd = [sys.executable, '-c', code]
+
+        max_args_length = 80
```

Comment:
This part is testing the specific implementation details rather than the general requirements. It clear to see this in this case, where you've had to replicate code from the implementation in order to write the test this way.

It is important to avoid this so that we don't have to update the test for every implementation change - that makes the test increase the burden of maintenance rather than reduce it.

To avoid this: make the test much less strict! For example, it could be enough check that the reprs are always start with "<Popen", end with ">", include the return code and at least part of the args which isn't too long (properly escaped). As for the length limit, IMO that's an implementation detail that we should *not* test at all.